src/Uno.UWP/Data/Pdf/PdfDocument.Android.cs

```diff
@@ -0,0 +1,112 @@
+		{
+			throw new ArgumentOutOfRangeException(nameof(pageIndex), $"In this document the page index cannot be {pageIndex}. Page count is {_pdfRenderer.PageCount}");
+		}
+
+		var pdfPage = _pdfRenderer.OpenPage((int)pageIndex);
+		return new PdfPage(pdfPage);
+	}
+
+	public static IAsyncOperation<PdfDocument> LoadFromFileAsync(Storage.IStorageFile file)
+	{
+		return LoadFromFileAsync(file, null);
+	}
+
+	public static IAsyncOperation<PdfDocument> LoadFromFileAsync(Storage.IStorageFile file, string password)
+	{
+		if (!string.IsNullOrEmpty(password))
+		{
+			// password protected PDF files are not supported by PdfRenderer in Android
+			throw new NotImplementedException("The member IAsyncOperation<PdfDocument> PdfDocument.LoadFromFileAsync(IStorageFile file, string password) is not implemented in Uno.");
+		}
+
+		if (file is null)
+		{
+			throw new ArgumentNullException(nameof(file));
+		}
+
+		var localpath = file.Path;
+		var fileDescriptor = localpath.StartsWith('/')
+			? ParcelFileDescriptor.Open(new Java.IO.File(localpath), ParcelFileMode.ReadOnly)
+			: Android.App.Application.Context.ContentResolver.OpenFileDescriptor(Android.Net.Uri.Parse(localpath), "r");
+		var pdfRenderer = new PdfRenderer(fileDescriptor);
+		var pdfDocument = new PdfDocument(pdfRenderer);
+
+		return Task.FromResult(pdfDocument).AsAsyncOperation();
+	}
+
+	public static IAsyncOperation<PdfDocument> LoadFromStreamAsync(Storage.Streams.IRandomAccessStream inputStream)
+	{
+		return LoadFromStreamAsync(inputStream, default);
+	}
+
+	public static IAsyncOperation<PdfDocument> LoadFromStreamAsync(Storage.Streams.IRandomAccessStream inputStream, string password)
+	{
+		if (!string.IsNullOrEmpty(password))
+		{
+			// password protected PDF files are not supported by PdfRenderer in Android
+			throw new NotImplementedException("The member IAsyncOperation<PdfDocument> PdfDocument.LoadFromStreamAsync(IRandomAccessStream inputStream, string password) is not implemented in Uno.");
+		}
+		return AsyncOperation.FromTask(async ct =>
+		{
+			var parcel = await GetParcelFileDescriptorFromStreamAsync(inputStream.AsStream(), ct);
+			var pdfRenderer = new PdfRenderer(parcel);
+			var pdfDocument = new PdfDocument(pdfRenderer);
+			return pdfDocument;
+		});
+	}
+
+	public void Dispose()
+	{
+		_pdfRenderer.Dispose();
+	}
+
+	private static Task<ParcelFileDescriptor> GetParcelFileDescriptorFromStreamAsync(Stream input, CancellationToken token) =>
+		Task.Run(() =>
+			{
+				/* PdfRenderer https://developer.android.com/reference/android/graphics/pdf/PdfRenderer#PdfRenderer(android.os.ParcelFileDescriptor)
+				 * PdfRenderer needs a seek-able ParcelFileDescriptor,
+				 * to get it from a stream (without using hacks),
+				 * you need to copy the stream to a temporary file.
+				*/
+				var fileName = Path.GetTempFileName();
+				var file = new Java.IO.File(fileName);
+				file.DeleteOnExit();
+				var outputStream = new Java.IO.FileOutputStream(file);
```

Comment:
If it is disposable (not sure), let's dispose it:

```suggestion
				using var outputStream = new Java.IO.FileOutputStream(file);
```